src/codeql.ts

```diff
@@ -60,7 +60,7 @@ libraryPathDependencies: codeql-${language}`
     queryFile,
   ]);
 
-  void Promise.all([
+  await Promise.all([
```

Comment:
@SamMorrowDrums without this change the code afterwards would sometimes run and find that these output files were not yet ready.

Reply:
Sorry that makes a lot more sense. It was an error on my part not to `await` - the `void` was to appease the linter but the real action should have been to actually handle the response by awaiting it ✔️

Reply:
🎉

Reply:
I agree that awaiting the promise is the right thing to do. I really like the linter check `no-floating-promises` which I guess this means you do have enabled.

There seems to be another case of using `void` to ignore the promise at
https://github.com/dsp-testing/qc-run2/blob/ac24ad467635caee2e47785a23a57180c978e694/src/combine-results.ts#L75
Although in this case we're not using the result of the promise (the worst that could happen is we say the results are available when they're not), it still looks dangerous to me to not wait for it.

Does it also stop errors from propagating? What happens if a promise that's being ignored like this fails?

Reply:
@robertbrignull yes - in all honesty there is zero reason not to wait, and as mentioned that means that it will throw an error rather than that weird state of unhandled failed promise in the ether.

I don't know what I was thinking when I read the lint - but there is absolutely reason to wait for both - and even a promise that you don't need to await should at minimum have a `.error(atLeastBloodyLogIt)` even if there is no reason to hold up execution of the subsequent code.